### ISSUE TITLE
Weird behaviour

### ISSUE THREAD

**Issue by busslina** [GregoryConrad/rearch-dart#153]
I created a simple dart test project to test and show this error.
I have a loading capsule that holds the app loading status. For now it only listen to the websocket state.
So, in the test, I simulated a Websocket connection flow.
You can see on the output, that this `LOADING SIDE EFFECT -- set() -- loading: false, message: null`, which is produced on `transactionRunner`, is not triggering a new update on the loading capsule listener.

Test repository: https://github.com/busslina/rearch-weird-behaviour

Output:
```
Starting loading state listener


WEBSOCKET: Disconnected


LOADING CAPSULE:
Websocket state: Disconnected
Loading: true
Message: Initiating

LOADING SIDE EFFECT -- set() -- loading: true, message: Disconnected

LISTENER:
Loading: true
Message: Initiating

Starting websocket simulation





WEBSOCKET: Connecting


LOADING CAPSULE:
Websocket state: Connecting
Loading: true
Message: Disconnected

LOADING SIDE EFFECT -- set() -- loading: true, message: Connecting

LISTENER:
Loading: true
Message: Disconnected





WEBSOCKET: Connected


LOADING CAPSULE:
Websocket state: Connected
Loading: true
Message: Connecting

LOADING SIDE EFFECT -- set() -- loading: true, message: Identifiying

LISTENER:
Loading: true
Message: Connecting





WEBSOCKET: Identified


LOADING CAPSULE:
Websocket state: Identified
Loading: true
Message: Identifiying

LOADING SIDE EFFECT -- set() -- loading: false, message: null

LISTENER:
Loading: true
Message: Identifiying
```

I'm expecting to get on the listener `loading --> false` and `message --> null`

**Comment by busslina** [comment]
I just think that if I return the loading value directly from the capsule, instead of mutating a side effect state, It will work.

So also the question is: I'm not allowed to do in that way? I think it should work.

**Comment by busslina** [comment]
> I just think that if I return the loading value directly from the capsule, instead of mutating a side effect state, It will work.
> 
> So also the question is: I'm not allowed to do in that way? I think it should work.

I just reproduced that thesis on a new branch and works as I expected:

Alternative repository: https://github.com/busslina/rearch-weird-behaviour/tree/alternative

**Comment by busslina** [comment]
> Rethinking it, I cannot understand why after a rebuild/transactionRunner, a dependant capsule is not being rebuild... (I'm using now a fully declarative approach that works well).

It just come to my mind that maybe the recent addition of `use.data` and the changes it involved maybe prevent to update dependent capsules (*) when rebuilding within the build.

(*) Maybe not for all dependant capsules but only for the one which is executing that rebuild.

**Comment by busslina** [comment]
> It just come to my mind that maybe the recent addition of use.data and the changes it involved maybe prevent to update dependent capsules (*) when rebuilding within the build.

> (*) Maybe not for all dependant capsules but only for the one which is executing that rebuild.

Seems true: https://github.com/GregoryConrad/rearch-dart/blob/09bdb4425eac81d77d7bb6eb3e9ad19eae554dfe/packages/rearch/lib/src/impl.dart#L72

**Comment by GregoryConrad** [comment]
Sorry, a little confused. What’s the actual vs expected behavior here?

**Comment by busslina** [comment]
I think I resolved my doubt looking at the changes you did when you introduced use.data

In few words I was expecting a capsule to rebuild after mutating one side effect of it. But I understood that any self state mutation executed from within the capsule will not cause rebuild.

Now, I refactored my loading capsule and it works fine.


Also, now I'm in another battle, trying to have a full warmup manager side effect that receives a Capsule reader and manages special data capsules throughout the full app lifecycle. It's supposed to handle data loading and data clearing. I suppose that I will create an issue to show/discuss it.
